### DISCUSSION TITLE
Convert ffmpeg filter to use it with mpv

### DISCUSSION THREAD

**Discussion by Ashyni** [Ashyni/mpv-scripts#24] [Off Topic]
Hey @geextahslex,
I created this topic to talk in a more appropriate place about your filter.

Here is a simple Lua script to try your filter with mpv:
```lua
local function on_start()
    mp.commandv("vf", "pre", "@myscript-label:lavfi=[split[original][denoised];[denoised]atadenoise[denoised_out];[original]lutyuv='u=128:v=128'[gray];[gray][denoised_out]blend=all_expr='if(between(A,0,25),B,if(between(A,25,35),B+(35-A)*(B-A)/10,A))']")
end

mp.register_event("file-loaded", on_start)
```

**Comment by geextahslex** [comment]
Thank you, that looks awesome.
I was experimenting with the command since yesterday and the command looks now like this
`-filter_complex "[0:V:0]split=3[unfiltered][denoised][mask];[mask]lutyuv=y=between(val\,0\,150)*maxval[mask];[denoised]atadenoise=s=5[denoised];[denoised][mask]mergeplanes=mapping=0x00010210:format=yuva420p10le[maskeddenoised];[unfiltered][maskeddenoised]overlay[out]" `

I tried to update it into the script you made but that failed ^^'

**Reply by Ashyni** [reply]
You need to escape `\`(ffmpeg) like that `\\`(mpv/lua).
```lua
mp.commandv("vf", "pre", "@myscript-label:lavfi=[[0:V:0]split=3[unfiltered][denoised][mask];[mask]lutyuv=y=between(val\\,0\\,150)*maxval[mask];[denoised]atadenoise=s=5[denoised];[denoised][mask]mergeplanes=mapping=0x00010210:format=yuva420p10le[maskeddenoised];[unfiltered][maskeddenoised]overlay[out]]")
```

**Reply by geextahslex** [reply]
Okay there is somehow no effect visible

**Reply by Ashyni** [reply]
my copy/pasta had some typo issue.
The filter is running, but it's hard so see the result compared the first post.

```lua
mp.commandv("vf", "pre",
        "@myscript-label:lavfi=[split=3[original][denoise][mask];[mask]lutyuv=y=between(val\\,0\\,150)*maxval[masked];[denoise]atadenoise=s=5[denoised];[denoised][masked]mergeplanes=0x00010210:yuva420p10le[merged];[original][merged]overlay]")
```

**Reply by geextahslex** [reply]
Awesome. I tried it like this, this also worked:
`mp.commandv("vf", "pre", "@myscript-label:lavfi=[vid1]split=3[unfiltered][denoised][mask];mask]lutyuv=y=between(val\\,0\\,150)*maxval[mask];[denoised]atadenoise[denoised];[denoised]mask]mergeplanes=mapping=0x00010210:format=yuva420p10le[maskeddenoised];[unfiltered][maskeddenoised]overlay[vo]]")`

Is there a way to toggle this via keybind? So it doesn't run on startup? ^^

**Reply by Ashyni** [reply]
Something like that, the keybind is `C` (Shift + c), change it on last line.
```lua
local label_prefix = mp.get_script_name()
local toggle = true

local function on_toggle()
    if toggle then
        mp.commandv("vf", "pre", string.format(
            "@%s:lavfi=[split=3[original][denoise][mask];[mask]lutyuv=y=between(val\\,0\\,150)*maxval[masked];[denoise]atadenoise=s=5[denoised];[denoised][masked]mergeplanes=0x00010210:yuva420p10le[merged];[original][merged]overlay]",
            label_prefix))
    else
        mp.commandv("vf", "remove", string.format("@%s", label_prefix))
    end
    toggle = not toggle
end

mp.add_key_binding("C", "toggle_filter", on_toggle)

```

**Reply by geextahslex** [reply]
Awesome, thank you :)

**Comment by geextahslex** [comment]
Hi, do you know why when trying to expand the lutyuv mask to `lutyuv=y=between(val\,0\,150)*1023+between(val\,151\,170)*(170-val)*10.23` to get a smooth transition, this doesn't work and gives an error, but in ffmpeg this works fine

I changed `atadenoise=s=5` with `hue=h=90:s=1` to make it easier to see what is happening

the script
```
local label_prefix = mp.get_script_name()
local toggle = true

local function on_toggle()
    if toggle then
        mp.commandv("vf", "pre", string.format(
            "@%s:lavfi=[split=3[original][denoise][mask];[mask]lutyuv=y=between(val\,0\,150)*1023+between(val\,151\,170)*(170-val)*10.23[masked];[denoise]hue=h=90:s=1[denoised];[denoised][masked]mergeplanes=0x00010210:yuva420p10le[merged];[original][merged]overlay]",
            label_prefix))
    else
        mp.commandv("vf", "remove", string.format("@%s", label_prefix))
    end
    toggle = not toggle
end

mp.add_key_binding("F2", "toggle_filter", on_toggle)
```

![mask error](https://github.com/user-attachments/assets/bf787c79-5823-4aeb-915c-d29217a928d0)

this is with ffmpeg only, you can see the green hue change
![ffmpeg](https://github.com/user-attachments/assets/ccadbe52-52b5-4829-8875-44e5554a362e)
command used in ffmpeg
`ffmpeg -i trek.mkv -filter_complex "[0:V:0]split=3[unfiltered][denoised][mask];[mask]lutyuv=y=between(val\,0\,150)*1023+between(val\,151\,170)*(170-val)*10.23[mask];[denoised]hue=h=90:s=1[denoised];[denoised][mask]mergeplanes=mapping=0x00010210:format=yuva420p10le[maskeddenoised];[unfiltered][maskeddenoised]overlay[out]" -map [out] -pix_fmt yuv420p10le -c:v libx265 -x265-params crf=20 -profile:v main10 changed2s60.mkv`

**Reply by Ashyni** [reply]
> You need to escape `\`(ffmpeg) like that `\\`(mpv/lua).

```lua
-- ffmpeg
lutyuv=y=between(val\,0\,150)*1023+between(val\,151\,170)*(170-val)*10.23

-- lua (mpv)
lutyuv=y=between(val\\,0\\,150)*1023+between(val\\,151\\,170)*(170-val)*10.23
```

**Reply by geextahslex** [reply]
That was a quick fix, thank you so much :)

**Comment by geextahslex** [comment]
Hi, I have a question. Do you know how to set this up to run with a profile-condition? So know I have to press my keybind, would it be possible to activate atadenoise when a profile-condition is met?
